feat: posting list

u00938 opened the pull request:
- Fixes #4
- Fixes #5 

- 게시물 목록 api (get: /postings/list) 
- 게시물 공유 api (put: /postings/share)

이상 있거나 빠진 곳 있으면 공유 부탁드립니다.

u00938 committed:
feat: 게시물 목록, 공유 기능 개발

u00938 committed:
fix: formatter, lint

jimin52 commented:
openapi 실행되나요? 제 환경에서는 에러가 발생합니다.

u00938 commented:
어떤 에러인가요?

u00938 commented:
혹시 다른 분들도 에러 나신다면 자세한 공유 좀 부탁드립니다!!!

scarf005 commented:
라우트에서 HTTP response 선언시 schema.ts의 zod 스키마를 사용해 주셨는데, 이는 DTO가 아닌 DB 엔티티 객체기 때문에 오류가 발생한 것으로 보입니다

u00938 committed:
fix: route type error

scarf005 committed:
fix: 경로에 `{id}` 파라미터 추가

Hono에서 path 문자열을 파싱해 param 타입을 생성하므로, 제거된 `{id}`을 복구했습니다.

scarf005 committed:
fix: `exactOptionalPropertyTypes` 비활성화

`z.optional()`과  `Hono`의 `c.req.valid`와 충돌을 일으켜 임시로 비활성화했습니다.

scarf005 committed:
fix: posting 목록 타입 오류 해결

- enum 사용
- optional 타입 사용
- 요구사항에 맞게 query 파라미터명 변경
- query 타입 추가
- 반환값 object 타입 제약 조건 제거
- JSON 파싱 플러그인 사용

scarf005 committed:
fix: posting share 타입 오류 해결

scarf005 reviewed (approved):
![image](https://github.com/7rd-wanted-pre-onboarding-team-H/w1-social-feed/assets/54838975/c38f686e-788e-41dd-86ea-d96460971586)
![image](https://github.com/7rd-wanted-pre-onboarding-team-H/w1-social-feed/assets/54838975/d18007bd-8aa0-4aab-a7fc-2467e2b35164)

수고 많으셨습니다!
